JoinColony/colonyCDapp
# [Multi-sig] feat: finalize a multi sig motion

## Description

This PR implements the option to finalize a multi sig motion. 
[Block ingestor PR](https://github.com/JoinColony/block-ingestor/pull/233)

## Testing

You will need 2 browsers, so invite `amy` or whoever you wish to `planex`.
For checking multi sig motion state, use this handy query
```
query MyQuery {
  getColonyAction(
    id: "THE_TX_HASH"
  ) {
    multiSigId
    id
    multiSigData {
      isExecuted
      isSuccess
      isRejected
      nativeMultiSigId
      signatures {
        items {
          multiSigId
          role
          userAddress
          vote
        }
      }
    }
    type
    isMultiSig
  }
}
```

1. Install the multi sig extension and set the global threshold to 2 (easiest to test). (you may need to refresh the app so multi sig pops up as decision method)
2. Assign the other user `Owner` roles with the authority of `Take actions via Multi-Sig`
3. Create a mint tokens action with the decision method set to `Multi-sig`
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/e98af115-2323-4c8a-8d2a-064999c9f023)
4. Copy over the url to the second browser, where you need to be logged in with `amy`
5. Click `Approve`, after a few moments the vote should show up on the right
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/b09f12d3-4c50-4d15-9f64-5c5ee75c3b59)
6. After the threshold has passed, the `Finalize` button should appear below the voting button. You can verify the votes and the state by running the gql query.
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/7cb23d14-023b-4428-b203-07e63655dbde)
7.  Click the `Finalize` button, after some moments, the transaction should pass, the sidebar should change to reflect that the multi sig has been finalized. You can run the query to make sure :)
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/6e0c64cb-7471-4c40-a38d-3d7f170cbb01)

Additionally check that the following voting reputation motions can be created normally:
1. Mint tokens
2. Unlock native token
3. Upgrade colony version

No need to go through staking, just the creating is enough.

## Diffs

**New stuff** ✨

* Added `isSuccess` and `executedAt` to the gql multisig schema
* Added mock `Finalize` button to finalize a multiSig
* `useDomainThreshold` which returns a multiSig's required threshold
* `useEligibleSignees` which returns all the role holders which can sign (this is going to be used for the widget, that's why it's a separate hook and not in the domain one)
* finalize multi sig saga

**Changes** 🏗
Block ingestor changes are documented in that PR because there's a lot of em :D
* Made it so that we use the "UI" roles for determining the permissions needed in the sagas
* permission utils naming tweaked
* when installing the extension, we give root users ALL the roles, not just root role
* refactored `rootMotion` saga and removed the `createRootMultiSig` saga, since all the action logic is the same, just some params are different
* realigned getting the payload for mint tokens, unlock native token and upgrade colony version actions

Resolves  #2157
